web/src/features/panels/zone/ZoneDetails.tsx

```diff
@@ -26,6 +36,19 @@ export default function ZoneDetails(): JSX.Element {
     return <Navigate to="/" replace />;
   }
 
+  useEffect(() => {
+    if (hasSubZones === null) {
+      return;
+    }
+    // When first hitting the map (or opening a zone from the ranking panel),
+    // set the correct matching view mode (zone or country).
+    if (hasSubZones && isZoneView) {
+      setViewMode(SpatialAggregate.COUNTRY);
+    } else if (isSubZone && !isZoneView) {
```

Comment:
Maybe a little nitpicky but I find an if statement without the else a little easier to read